### PR TITLE
Merge GithubException.pyi stubs back to source

### PR DESCRIPTION
**Pull request by trim21**
ref https://github.com/PyGithub/PyGithub/issues/2462

### CODE REVIEW

File: github/GithubException.py [hunk 1]
```diff
@@ -38,35 +39,35 @@ class GithubException(Exception):
...
     """
 
-    def __init__(self, status, data, headers):
+    def __init__(self, status: int, data: Any, headers: Optional[Dict[str, str]]):
         super().__init__()
         self.__status = status
         self.__data = data
         self.__headers = headers
-        self.args = [status, data, headers]
+        self.args = (status, data, headers)
```
Comment (trim21) [inline github/GithubException.py:47]: type of `Exception.args` is `Tuple[Any, ...]` so mypy will raise a error for this

Comment (trim21) [inline github/GithubException.py:47]: type of `Exception.args` is `Tuple[Any, ...]` so mypy will raise a error for this

**Comment by EnricoMi** [comment]
Would you prefer to also improve the typing or only merge the `.pyi` into the `.py` file?

**Comment by trim21** [comment]
> Would you prefer to also improve the typing or only merge the `.pyi` into the `.py` file as is?

also improve the typing, we may find type error in method function body when merging types back

**Comment by trim21** [comment]
do you perfer `from typing import ...` or `import typing as t` ?

**Comment by EnricoMi** [comment]
> do you perfer `from typing import ...` or `import typing as t` ?

The `.pyi` files use the former throughout this project, so I would go for that.

File: github/GithubException.py [hunk 2]
```diff
@@ -38,35 +39,35 @@ class GithubException(Exception):
...
-    def data(self):
+    def data(self) -> Dict[str, Union[str, List[str], List[Dict[str, str]]]]:
         """
         The (decoded) data returned by the Github API
         """
         return self.__data
 
     @property
-    def headers(self):
+    def headers(self) -> Union[None, Dict[str, str]]:
```
Comment (EnricoMi) [inline github/GithubException.py:64]: I think `Optional[T]` is better than `Union[None, T]`.

File: github/GithubException.py [hunk 3]
```diff
@@ -99,27 +100,44 @@ class BadAttributeException(Exception):
     Exception raised when Github returns an attribute with the wrong type.
     """
 
-    def __init__(self, actualValue, expectedType, transformationException):
+    def __init__(
+        self,
+        actualValue: Any,
+        expectedType: Union[
```
Comment (EnricoMi) [inline github/GithubException.py:106]: Is this type really correct / complete?

File: github/GithubException.py [hunk 4]
```diff
@@ -38,35 +39,35 @@ class GithubException(Exception):
...
-    def status(self):
+    def status(self) -> int:
         """
         The status returned by the Github API
         """
         return self.__status
 
     @property
-    def data(self):
+    def data(self) -> Dict[str, Union[str, List[str], List[Dict[str, str]]]]:
```
[the remark this answers is not in the document]
Comment (trim21) [reply github/GithubException.py:57]: I'm not 100% confident about these types, these are copied from pyi, I'm not sure how it's used,

[the remark this answers is not in the document]
Comment (trim21) [reply github/GithubException.py:62]: we can fix this later, when they are used in source, I think

**Review by EnricoMi — APPROVED** [review]

**Pull request merged**